zebrad/tests/acceptance.rs

```diff
@@ -1467,6 +1467,52 @@ async fn rpc_endpoint(parallel_cpu_threads: bool) -> Result<()> {
     Ok(())
 }
 
+/// Test that the JSON-RPC endpoint responds to a request even if the content type is not json.
+///
+/// This test ensures that the examples of zcashd rpc methods will also work in Zebra.
+///
+/// https://zcash.github.io/rpc/getblockchaininfo.html
+#[tokio::test]
+async fn rpc_endpoint_ignore_client_content_type() -> Result<()> {
+    let _init_guard = zebra_test::init();
+    if zebra_test::net::zebra_skip_network_tests() {
+        return Ok(());
+    }
+
+    // Write a configuration that has RPC listen_addr set
+    // [Note on port conflict](#Note on port conflict)
+    let mut config = random_known_rpc_port_config(true)?;
+
+    let dir = testdir()?.with_config(&mut config)?;
+    let mut child = dir.spawn_child(args!["start"])?;
+
+    // Wait until port is open.
+    child.expect_stdout_line_matches(
+        format!("Opened RPC endpoint at {}", config.rpc.listen_addr.unwrap()).as_str(),
+    )?;
+
+    // Create an http client
+    let client = RpcRequestClient::new(config.rpc.listen_addr.unwrap());
+
+    // Make the call to the `getinfo` RPC method with a `text/plain` content type as the zcashd rpc docs.
+    let res = client
+        .call_with_content_type("getinfo", "[]".to_string(), "text/plain".to_string())
+        .await?;
+
+    // Test if rpc endpoint respond
+    assert!(res.status().is_success());
+
+    // Make the call to the `getinfo` RPC method with a random string as content type.
+    let res = client
+        .call_with_content_type("getinfo", "[]".to_string(), "whatever".to_string())
```

Comment:
I don't think we want to allow any content type value, because that would allow some kinds of cross-application attacks.

This is a potential security issue.

Can we only allow:
- no content-type
- text/plain
- anything else we can find in actual clients or documentation examples?

Reply:
This was changed at https://github.com/ZcashFoundation/zebra/pull/6885/commits/6ef6969397f44c560c7c772461f7ccbc0fb6e1c9